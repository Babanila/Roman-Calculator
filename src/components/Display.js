import React from "react";
import styled from "@emotion/styled";

const Output = styled.div`
    height: 5em;
    grid-column: 1 / -1;
    background-color: rgba(0, 0, 0, 0.75);
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    align-items: flex-end;
    justify-content: space-around;
    padding: 10px;
    word-wrap: break-word;
    word-break: break-all;
    z-index: 70;
`;

const Previous = styled.div`
    color: rgba(255, 255, 255, 0.75);
    font-size: 1.5rem;
`;

const Current = styled.div`
    color: white;
    font-size: 2.5rem;
`;

const Display = props => {
    const { prevOperation, currentOperation, operator } = props;

    return (
        <Output>
            <Previous data-previous-operand>
                {prevOperation} {operator}
            </Previous>
            <Current data-current-operand>{currentOperation}</Current>
        </Output>
    );
};

export default Display;
