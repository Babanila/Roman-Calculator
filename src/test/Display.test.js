import React from "react";
import renderer from "react-test-renderer";
import { render, cleanup } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import Display from "../components/Display";

describe("Display Component Test", () => {
    afterEach(cleanup);

    let prevOperation, currentOperation, operator;

    beforeEach(() => {
        prevOperation = "XL";
        currentOperation = "VI";
        operator = "+";
    });

    it("should renders without crashing", () => {
        const DisplayComponent = renderer
            .create(
                <Display
                    prevOperation={prevOperation}
                    currentOperation={currentOperation}
                    operator={operator}
                />
            )
            .toJSON();
        expect(DisplayComponent).toMatchSnapshot();
    });

    it("should render correct number of element", () => {
        const DisplayWrapper = shallow(
            <Display
                prevOperation={prevOperation}
                currentOperation={currentOperation}
                operator={operator}
            />
        );
        expect(DisplayWrapper.find("div").exists).toBeTruthy();
        expect(DisplayWrapper.find("div").length).toEqual(0);
    });

    it("should render correct number of children element", () => {
        const DisplayWrapper = mount(
            <Display
                prevOperation={prevOperation}
                currentOperation={currentOperation}
                operator={operator}
            />
        );
        expect(DisplayWrapper.find("div").exists).toBeTruthy();
        expect(DisplayWrapper.find("div").length).toEqual(3);
    });

    it("should show the right values for output", () => {
        const { getByText } = render(
            <Display
                prevOperation={prevOperation}
                currentOperation={currentOperation}
                operator={operator}
            />
        );

        expect(getByText("XL +")).toBeInTheDocument();
        expect(getByText("VI")).toBeInTheDocument();
    });
});
