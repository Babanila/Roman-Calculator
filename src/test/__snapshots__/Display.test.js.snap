// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Component Test should renders without crashing 1`] = `
<div
  className="css-vb37s6"
>
  <div
    className="css-nrsn95"
    data-previous-operand={true}
  >
    XL
     
    +
  </div>
  <div
    className="css-1jxc83g"
    data-current-operand={true}
  >
    VI
  </div>
</div>
`;
